### ISSUE TITLE
Semaphore - Might need disclaimer

### ISSUE THREAD

**Issue by GantMan** [danger/danger-js#457]
Not sure where the docs for this page are located, otherwise I would have just done a PR:
http://danger.systems/js/guides/getting_started.html

The semaphore integration works, but there's a catch.  Semaphore won't do a PR build unless it's a fork.  So anyone who owns the repo, and simply does branch PRs, will see this message:
![image](https://user-images.githubusercontent.com/997157/34343875-533b7d78-e9a3-11e7-8733-5cf23747def2.png)

I wrote to Semaphore and they responded with the following:
```
After a project has been added, Semaphore builds branches that are created within the repository.
 Because of this, at the moment Semaphore doesn't support building pull requests that are opened 
within the same repository. Opening a PR will trigger a build only if it was opened from a fork. For 
more details please refer to this page. 

Good news is that adding the option for building pull requests within repository is on our roadmap. 
So, we'd really appreciate if you have few moments to describe your use-case. We don't have an 
ETA yet, but we expect to deliver this feature during the first half of 2018. 
```

I believe a caveat should be added to the docs (I personally moved danger-js to Travis for this reason).  But then again, maybe danger JS should detect semaphore and be willing to skip the detection process?

I hope this helps ^

**Comment by orta** [comment]
The code comes from here: https://github.com/danger/danger-js/blob/master/source/ci_source/providers/Semaphore.ts#L4-L16


Danger doesn't strictly check if it's a PR, but cares about [these build vars](https://github.com/danger/danger-js/blob/9772aacbb62155a55ee5e6a1e4346ebb9aea0129/source/ci_source/providers/Semaphore.ts#L27-L31) - could there be alternatives?

**Comment by GantMan** [comment]
Yeah, there's not going to be a `PULL_REQUEST_NUMBER` if semaphore is simply treating it as a branch.   I assume that's the crux of the problem.

### Also Merry Xmas!  🎄 🎁  🎅 🤶

**Comment by orta** [comment]
same 🥇

**Comment by Elecweb** [comment]
Have any updated ? I also have this problem. Do you haven workaround for this one ?

**Comment by bessey** [comment]
Here's your workaround https://semaphoreci.com/docs/how-to-fetch-pull-request-number-from-github.html
